include/pisa/query/queries.hpp

```diff
@@ -92,12 +92,13 @@ struct Query {
 
 [[nodiscard]] std::function<void(const std::string)> compute_parse_query_function(
     std::vector<Query> &queries,
-    const std::optional<std::string> terms_file,
-    const std::optional<std::string> stopwords_filename,
-    const std::optional<std::string> stemmer_type) {
+    std::optional<std::string> const &terms_file,
+    std::optional<std::string> const &stopwords_filename,
+    std::optional<std::string> const &stemmer_type)
+{
     if (terms_file) {
-        auto term_processor = TermProcessor(terms_file, stopwords_filename, stemmer_type);
-        return [&](std::string const &query_line) {
+        return [&, term_processor = TermProcessor(terms_file, stopwords_filename, stemmer_type)](
```

Comment:
Sorry for this bug. This is due to `term_processor` is destroyed after exiting of the function, isn't it? In the other hand, looking again, would it be an adventage capturing only `&queries` instead `&`?